inc/Engine/Optimization/RUCSS/Warmup/Subscriber.php

```diff
@@ -0,0 +1,79 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Optimization\RUCSS\Warmup;
+
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+class Subscriber implements Subscriber_Interface {
+
+	/**
+	 * Options instance.
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Resource object.
+	 *
+	 * @var ResourceFetcher
+	 */
+	private $resource_fetcher;
+
+	/**
+	 * Subscriber constructor.
+	 *
+	 * @param Options_Data    $options Options instance.
+	 * @param ResourceFetcher $resource_fetcher Resource object.
+	 */
+	public function __construct( Options_Data $options, ResourceFetcher $resource_fetcher ) {
+		$this->resource_fetcher = $resource_fetcher;
+		$this->options          = $options;
+	}
+
+	/**
+	 * Return an array of events that this subscriber listens to.
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() : array {
+		return [
+			'rocket_buffer' => 'collect_resources',
```

Comment:
We need to discuss the priority for `collect_resources`.
Collect Resources should happen before TreeShake.
On `rocket_buffer` we have a couple of Optimizations which do not need to be sent to TreeShake ( SAAS expects the HTML without minified CSS, JS ). 
In Phase 3 I've sent TreeShake on priority 1, however, I think collect_resources should be 1 & TreeShake should be 2.

Reply:
@crystinutzaa 
Good question, really I don't like using 1 and 2 we can start from 10 so other plugins (or maybe we) can hook it before both of them.
I think we can make warmup at first thing (priority 10) and let treeshaker comes at the last thing of optimizations so it will give some time to warmup to finish his work, 

what do u think?

Reply:
`collect_resources` should happen very early. Maybe priority 5 to allow for other things to get in first, but earlier than 10, because we don't want other things with default 10 to fire before.
I think we can't do treeshake last, b/c of @crystinutzaa's observation that some optimizations will happen after we have the shaken CSS (minify, specifically, maybe others). So it should be late, but before any of those.

Reply:
I'm favoring a bigger priority gap between collect and optimize here, because it allows a few more ms for the SaaS warmup to collect maybe a few more resources before the call to treeshake hits the SaaS. It _may_ provide a slightly better outcome on those first few calls, which would be better early result for the customer and fewer re-queued resources for the next time warmup runs. 🤔

Reply:
Hem....TreeShake need to kick in before any HTML optimizations. So it needs to be run before applying CDN, Minify CSS, Minify JS , lazyload. All of these start from priority 10 (some are on 14, 16,17). 
So, I assume Collect Resources will go on 10 and TreeShake on 11. I don't think we have other option.

Reply:
Collect on 4, shake on 9?

Reply:
Yeah. I think collect on 4 and Shake on 11. However, imagine that nothing else is running in between if we will set it on 4 and 9. 😅 
On priority 10, we run `cache_busting_facebook_tracking` , `cache_busting_google_tracking`, 
On Priority 1, we run: `extract_ie_conditionals`